library/std/src/io/stdio.rs

```diff
@@ -1202,6 +1202,59 @@ where
     }
 }
 
+/// Trait to determine if stdio stream (Stdin, Stdout, Stderr) is a terminal/tty.
+#[unstable(feature = "is_terminal", issue = "80937")]
+pub trait IsTerminal {
+    /// returns true if stdio stream (Stdin, Stdout, Stderr) is a terminal/tty.
+    fn is_terminal() -> bool;
```

Comment:
```suggestion
    fn is_terminal() -> io::Result<bool>;
```

This would allow for implementations that might return errors when attempting to determine this.